witchcraft/service_discovery.go

```diff
@@ -46,15 +46,22 @@ type ConfigurableServiceDiscovery interface {
 	// to benefit from live-reloadability. All config provided by WithServiceConfig will be merged into the default
 	// refreshable ServicesConfig. In case of conflict, the config provided here overrides the defaults.
 	WithServiceConfig(serviceName string, cfg httpclient.ClientConfig)
+	// WithDefaultParams stores the provided params function. When constructing a new client, the function will
+	// be evaluated and the params added to all clients.
+	WithDefaultParams(params func(serviceName string) ([]httpclient.ClientParam, error))
```

Comment:
Can you mention that this can be called multiple times or something similar to make it clear the params are additive rather than overwrite an existing param func. The ordering for how the params are applied might also be relevant since params added last can overwrite a previous param.